src/sentry/migrations/0422_backfill_denormalize_notification_actor.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 2.2.28 on 2023-04-17 16:12
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+ACTOR_TEAM_TYPE = 0
+ACTOR_USER_TYPE = 1
+
+
+def backfill_denormalize_notification(apps, schema_editor):
+    NotificationSetting = apps.get_model("sentry", "NotificationSetting")
+    for ns in RangeQuerySetWrapperWithProgressBar(
+        NotificationSetting.objects.select_related("target").all()
+    ):
+        target = ns.target
+        if target:
+            if target.type == ACTOR_TEAM_TYPE and target.team_id and not ns.team_id:
+                ns.update(team_id=target.team_id)
+            elif target.type == ACTOR_USER_TYPE and target.user_id and not ns.user_id:
+                ns.update(user_id=target.user_id)
+
+
+class Migration(CheckedMigration):
+    # This flag is used to mark that a migration shouldn't be automatically run in production. For
+    # the most part, this should only be used for operations where it's safe to run the migration
+    # after your code has deployed. So this should not be used for most operations that alter the
+    # schema of a table.
+    # Here are some things that make sense to mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that they can
+    #   be monitored and not block the deploy for a long period of time while they run.
+    # - Adding indexes to large tables. Since this can take a long time, we'd generally prefer to
+    #   have ops run this and not block the deploy. Note that while adding an index is a schema
+    #   change, it's completely safe to run the operation after the code has deployed.
+    is_dangerous = False
```

Comment:
I think we'll need this to be True. Then we can open an ops ticket to run it once its merged

Reply:
This will be run after https://github.com/getsentry/sentry/pull/47118 merges